### ISSUE TITLE
`allow_attributes` triggers on variable name

### ISSUE THREAD

**Issue by Byter09** [rust-lang/rust-clippy#13380]
### Summary

When upgrading to 1.81.0 the lint triggered on a variable in our code. I managed to produce a minimal example, unsure if it can be even smaller.

### Lint Name

allow_attributes

### Reproducer

I tried this code:

```rust
#![deny(clippy::allow_attributes)]


pub fn minimal_example() -> Option<u8> {
    let allowed_opt: Option<u8> = None;

    let allowed_raw = allowed_opt?;

    Some(allowed_raw)
}
```

I saw this happen:

```
    Checking playground v0.0.1 (/playground)
error: #[allow] attribute found
 --> src/lib.rs:7:23
  |
7 |     let allowed_raw = allowed_opt?;
  |                       ^^^^^^^^^^^^ help: replace it with: `expect`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#allow_attributes
note: the lint level is defined here
 --> src/lib.rs:1:9
  |
1 | #![deny(clippy::allow_attributes)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^

error: could not compile `playground` (lib) due to 1 previous error
```

I expected to see this happen:

```
    Checking playground v0.0.1 (/playground)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 1.47s
```

### Version

```text
rustc 1.81.0 (eeb90cda1 2024-09-04)
binary: rustc
commit-hash: eeb90cda1969383f56a2637cbd3037bdf598841c
commit-date: 2024-09-04
host: x86_64-unknown-linux-gnu
release: 1.81.0
LLVM version: 18.1.7
```

**Comment by samueltardieu** [comment]
Thanks for the report @Byter09. This should be fixed in Rust/Clippy 1.83, as it was due to a bug in the rust compiler and has been fixed for 1.83.

**Comment by Byter09** [comment]
Neat. Thanks for the quick fix. :D
